src/group_theory/nilpotent.lean

```diff
@@ -690,6 +691,49 @@ begin
   exact upper_central_series_eq_top_iff_nilpotency_class_le.mpr h,
 end
 
+section prod
+
+variables {G₁ : Type*} {G₂ : Type*} [group G₁] [group G₂]
+
+@[simp]
```

Comment:
I don't think this should be `simp`.

I can imagine an argument about a group which happens to be a product but for which this is incidental. In such a situation we would not want `simp` to perform this substitution.